### PR TITLE
Fix `multi_gpu_model` errors

### PR DESCRIPTION
**Pull request by taehoonlee**
This PR addresses `multi_gpu_model` errors described in #8976. The example codes are:
```python
import numpy as np
import tensorflow as tf
from keras.models import Sequential
from keras.layers import Dense, LSTM, Conv2D
from keras.utils import multi_gpu_model

with tf.device('/cpu:0'):
    model0 = Sequential()
    model0.add(LSTM(32, dropout=0.2, recurrent_dropout=0.2, input_shape=(24, 8)))

model = multi_gpu_model(model0, gpus=2)
model.compile(loss='mse', optimizer='adam')

x_train = np.random.random((200, 24, 8))
y_train = np.random.random((200, 32))

model.fit(x_train, y_train, epochs=1, batch_size=20)
```
And the results are:
```
   InvalidArgumentError (see above for traceback): Incompatible shapes: [10,128] vs. [20,128]
 [[Node: replica_0/sequential_1/lstm_1/while/mul_5 = Mul[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/gpu:0"](replica_0/sequential_1/lstm_1/while/Identity_2, replica_0/sequential_1/lstm_1/while/mul_5/Enter)]]
```

The error arises from the dropout masking (e.g., `h_tm1_i = h_tm1 * rec_dp_mask[0]`). The batch-size of the dropout mask (`rec_dp_mask[0]`) is determined when the CPU model is built, while the the one of the previous states (`h_tm1`) is done when the multi-GPU model is constructed.

Thus, one of the possible solutions is to release the dropout mask generated by the CPU model in order to make new dropout mask when `multi_gpu_model(model0, gpus=2)` is called.

### CODE REVIEW

**Comment by fchollet** [comment]
I think the issue is more general than that. Wouldn't it mean that a RNN layer with dropout couldn't be reused across different batch sizes?

In practice, the following seems to work:

```python
lstm = LSTM(3, dropout=0.2, recurrent_dropout=0.3)

a = Input(batch_shape=(3, 5, 10))
b = Input(batch_shape=(4, 6, 10))
c = lstm(a)
d = lstm(b)
```

**Comment by taehoonlee** [comment]
You are right @fchollet, a RNN layer with dropout can not be reused across different batch sizes. In addition to your code snippet, the example codes and results are:
```python
model1 = Model(a, c)
model1.compile(loss='mse', optimizer='adam')
model1.fit(x_train1, y_train1, epochs=1, batch_size=3)  # success
model2 = Model(b, d)
model2.compile(loss='mse', optimizer='adam')
model2.fit(x_train2, y_train2, epochs=1, batch_size=4)  # failure
```

```
InvalidArgumentError (see above for traceback): Incompatible shapes: [3,3] vs. [4,3]
	 [[Node: lstm_1_1/while/mul_5 = Mul[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/gpu:0"](lstm_1_1/while/Identity_2, lstm_1_1/while/mul_5/Enter)]]
```

**Comment by fchollet** [comment]
In this case we need to fix the underlying problem at the level of RNN layers, rather than patching it for `multi_gpu_model` only. My guess is that we can reset the dropout masks at every `call`.
